proposals/gc/Overview.md

```diff
@@ -574,7 +579,7 @@ Imported types are essentially parameters to the module.
 As such, they are entirely abstract, as far as compile-time validation is concerned.
 The only operations possible with them are those that do not require knowledge of their actual definition or size: primarily, passing and storing references to such types.
 
-TODO: The ability to import types makes the type and import sections interdependent.
+TODO: The ability to import types makes the type and import sections interdependent. (we will probably need to add placeholder entries in the types section for imported types, and possibly may need to add assumptions about their representation, even to compile code that uses them as locals, unless we want to restrict imported types to be only reference types).
```

Comment:
The placeholder idea was to reserve indices in the type space for imported types, but I guess we could do the same thing as function types and prepend the imported types to the type index space. I suppose the constraints could be expressed in the import.

Note that supertype bounds aren't really enough; one could view imported types as a poor man's type parameters, and generating code that manipulates imported types (even just for calling conventions) would require knowledge of the representation (thus the comment about assuming imports can only be reference types).